gtests/test_nopython_serializable.cpp

```diff
@@ -163,4 +167,111 @@ TEST(Json, serialization_with_object)
                          "\"pets\":[{\"name\":\"Fluffy\",\"age\":3,\"is_dog\":true,\"is_cat\":false},{\"name\":\"Whiskers\",\"age\":8,\"is_dog\":false,\"is_cat\":true}]}";
 }
 
+TEST(Json, deserialize_private_member_partial_exposure)
```

Comment:
We will need Python tests for these things.  It's not productive to test for strings in C++.

It can use a new PR or even a new issue.